lib/webmock/http_lib_adapters/manticore_adapter.rb

```diff
@@ -58,7 +55,20 @@ def join_array_values(headers)
           end
 
           def response_object_for(client, request, context, &block)
-            request_signature = generate_webmock_request_signature
+            if request.respond_to?(:entity) && !request.entity.nil?
+              body = Manticore::EntityConverter.new.read_entity(request.entity)
+            else
+              body = nil
+            end
```

Comment:
I'd prefer this pulled out into a separate method to keep `response_object_for` relatively high-level for readability. Also feels like there should be a better alternative to the `respond_to?` and `nil?` checks, but I'm not sure what off the top of my head.